system/lib/libcxxabi/src/cxa_guard.cpp

```diff
@@ -9,6 +9,29 @@
 
 #include "abort_message.h"
 
+#if __EMSCRIPTEN__
+
+// XXX EMSCRIPTEN making guard operations simple and LTO-optimizable opens up
```

Comment:
Yes and no. No, since this literally moved thread-unsafe guards from js into C, they were always not threadsafe. I think it's just not common at all to have threads started up in global ctors, which is the main user of cxa_guard stuff.

But yes, if we want to be more correct, we could add full threading support here. But as this isn't a regression, I suggest we consider it later separately.